### PR TITLE
Updated Vim document

### PR DESCRIPTION
**Pull request by zarej**
- Added line numbers in the configuration
- Removed nodejs installation part since it is already present in nix-shell
- Note in the beginning that nix-shell is using
- Better organized, fixed typos

### CODE REVIEW

**Review by nstankov-bg — APPROVED** [review]
Looks Great!
Will wait for @dino-  to approve as well.

File: book/docs/setup/editors/vim.md [hunk 1]
```diff
@@ -195,10 +226,17 @@ nnoremap <silent><nowait> <space>k  :<C-u>CocPrev<CR>
 " Resume latest coc list.
 nnoremap <silent><nowait> <space>p  :<C-u>CocListResume<CR>
 
+
+" Custom setings
```
Comment (dino-) [inline book/docs/setup/editors/vim.md]: Typo: "setings" should be "settings"

File: book/docs/setup/editors/vim.md [hunk 2]
```diff
@@ -212,11 +250,25 @@ Now you can enter to `vim` and execute `:PlugInstall` to install `coc.nvim` and
...
+We have `haskell-language-server` already installed and accessible via `nix-shell`.
+
+## Start Vim/Nvim
+Now you can use `vim` or `neovim` to edit Haskell files with code completion.
+
+When you enter the first time to vim it will need to wait few minutes until code completion is initialized and ghc and cabal finish their jobs. If you are using Neovim you will see progress in status bar.
+
+## Keyboars Shortcuts
+ 	
+\<leader\> key is ussually `\`
```
Comment (dino-) [inline book/docs/setup/editors/vim.md]: Typo: "ussually" should be "usually"

File: book/docs/setup/editors/vim.md [hunk 3]
```diff
@@ -195,10 +226,17 @@ nnoremap <silent><nowait> <space>k  :<C-u>CocPrev<CR>
...
+set number
+" tab and indentation setup
+set tabstop=8 softtabstop=0 expandtab shiftwidth=4 smarttab autoindent
+
 ```
-The first part is `coc.nvim` plugin and the rest is configuration for a code completition.
+The first part is `coc.nvim` plugin and the rest is configuration for a code completition. At the last we have `Custom settings` for the other useful configuration like line numbers and tab and indent setup.
 
-Now you can enter to `vim` and execute `:PlugInstall` to install `coc.nvim` and then to configure with `:CocConfig` by entering the following:
+Now you can enter to `vim` and execute `:PlugInstall` to install `coc.nvim`. After that you need to exit Vima and enter again to launch `coc-nvim` plugin. Next step is to configure it with vim command `:CocConfig` by entering the following:
```
Comment (dino-) [inline book/docs/setup/editors/vim.md]: Typo: "Vima" should be "Vim"

**Review by dino- — CHANGES_REQUESTED** [review]
Looks great other than 3 typos I noticed and commented on. Lines 230, 239 and 262

**Review by dino- — APPROVED** [review]
Thank you!

**Pull request merged**